src/UseCases/CancelDonation/CancelDonationUseCase.php

```diff
@@ -22,12 +23,30 @@ public function __construct( DonationRepository $donationRepository, TemplateBas
 	}
 
 	public function cancelDonation( CancelDonationRequest $cancellationRequest ): CancelDonationResponse {
+		try {
+			$donation = $this->donationRepository->getDonationById( $cancellationRequest->getDonationId() );
+		}
+		catch ( GetDonationException $ex ) {
+			return new CancelDonationResponse( $cancellationRequest->getDonationId(), false );
```

Comment:
It's still unclear to me what the response model should be (https://phabricator.wikimedia.org/T123046), so I'm not spending time on making this nice while not knowing if it's the right stuff to begin with